app/public/cantusdata/models/chant.py

```diff
@@ -25,6 +25,7 @@ class Meta:
     lit_position = models.CharField(max_length=255, blank=True, null=True)
     mode = models.CharField(max_length=255, blank=True, null=True)
     differentia = models.CharField(max_length=255, blank=True, null=True)
+    differentiae_database = models.CharField(max_length=255, blank=True)
```

Comment:
- You should try sizing these fields to the actual expected length of the data, instead of just copying it.
- `blank=True` generally also needs `null=True`. The former validates that forms in the admin interface can be blank, while the latter allows blank values in the database. It's not very often you would need one without the other.

Reply:
> * You should try sizing these fields to the actual expected length of the data, instead of just copying it.

Sounds good!

> * `blank=True` generally also needs `null=True`. The former validates that forms in the admin interface can be blank, while the latter allows blank values in the database. It's not very often you would need one without the other.

Sounds good....for my own understanding...my impression was that since blank CharFields and TextFields are stored as `""` they wouldn't be `null`...but it is still good practice to have `null = True` for these fields?

> * `differentia` vs. `differentiae_database`... I know latin plurals, blah blah blah, but it would be good to choose one form or another.

Yes...this is what CantusDB calls them, and I'm wary of changing the names here but not there? I also find them terribly confusing and I think we have to come up with a more clear way to distinguish them/say what they are for search (conversation ongoing in #804) (although of course we can refer to them differently/provide explanatory information in the front-end). At the very least, I think this would be a good place for some help text.

Reply:
> my impression was that since blank CharFields and TextFields are stored as "" they wouldn't be null...but it is still good practice to have null = True for these fields?

If a field is empty, it should be NULL. Python "truthiness" obscures this, but `""` is not the same as `None`. The ORM maps non-nullable empty fields to `""`, but NULL empty fields to `None`. 

Bugs can manifest themselves quite easily in your code, then, if you have one check that says:

```python
if model.field:
    ... do something
```

and another that says:

```python
if model.field is None:
     ... do something
```

With `null=True` then they are both the same, since it's either a value or `None`; with `null=False` they are not since they could be `None`, an empty string `""`, or a value.

Reply:
Ah, ok...this makes sense. Thanks!